gatoryaml/gator_yaml.py

```diff
@@ -1,106 +1,93 @@
-        for i in list_in:
-            if isinstance(i, dict):
-                self.tabs += 1
-                self.enum_dict(i)
-            else:
-                self.output_list_item(i)
-
-    def enum_dict(self, dic):
-        """Enumerate through input dictionary and output each key"""
-        self.tabs += 1
-
-        for k in dic.keys():
-            if k == "indent":
-                self.indents = int(dic[k])
-
-            if k == "files":
-                self.tabs -= 1
-                # if isinstance(d[k], list):
-                #     self.enum_list(d[k])
-                if isinstance(dic[k], dict):
-                    self.enum_file_dict(dic[k])
-            elif isinstance(dic[k], list):
-                self.output_key(k)
-                self.enum_list(dic[k])
-            elif isinstance(dic[k], dict):
-                self.output_key(k)
-                self.enum_dict(dic[k])
-            else:
-                if not self.is_keyword(k, dic[k]):
-                    self.output_key_value(k, dic[k])
-
-        self.tabs -= 1
-
-    def enum_file_dict(self, files):
-        """Enumerate through the file list dictionary and output each key"""
-        self.tabs += 1
-
-        for k in files:
-            if isinstance(files[k], dict):
-                self.output_key(k)
-                self.enum_file_dict(files[k])
-            elif isinstance(files[k], list):
-                self.output_key(k)
-                self.enum_file_list(files[k])
-
-        self.tabs -= 1
-
-    def enum_file_list(self, list_in):
-        """Enumerate through each file key's parameter list items"""
-        for item in list_in:
-            self.output += (" " * self.spaces) * self.indents + str(item) + "\n"
-
-    def output_list_item(self, item):
-        """Output a generic list item"""
-        self.output += (" " * self.spaces) * self.tabs + " -" + str(item) + "\n"
-
-    def output_key(self, key):
-        """Output a generic key"""
-        self.output += (" " * self.spaces) * self.tabs + str(key) + ":\n"
-
-    def output_key_value(self, key, value):
-        """Output a generic key and it's value"""
-        self.output += ((" " * self.spaces) * self.tabs) + str(key) + ": " + str(value) + "\n"
-
-    def is_keyword(self, key, value):
-        """Output key and value if a keyword"""
-        if key in self.keywords:
-            if key == "commits":
-                self.output += (" " * self.spaces) * self.tabs \
-                               + "--" + str(key) + " " + str(value) + "\n"
-            else:
-                self.output += (" " * self.spaces) * self.tabs + str(key) \
-                               + " " + str(value) + "\n"
-            return True
-        return False
+from .exceptions import UnexpectedValue
+
+
+def dump(header, body, indent=4, spaces=4) -> str:
+    """Input dictionary is parsed. returns string of valid YAML"""
+
```

Comment:
Should maybe think about setting `spaces` from the value of `indent` in `header` if it exists.

Reply:
So spaces is just how many spaces count as a tab. It *shouldn't* ever change from 4 but I figured it was worth it to have it as an option.

Reply:
Yes; this is exactly what the `indents` parameter in the header controls.

Reply:
If by tab you mean "single level of indentation"

Reply:
Ahh, okay I understand now. That was a misunderstanding on my part, I (for whatever reason) thought indent was supposed to define an indent level for the params regardless of the key's indent level. My bad.